### PR TITLE
[CosmosDB] Upgrade to version 3.0.0 and add support for NetworkAclBypass + Update Mongo ServerVersion + backup policy

### PR DESCRIPTION
**Pull request by kristynhamasaki**
**Description**

Change needed for ignite with support of NetworkAclBypass.
-Adds params network_acl_bypass and network_acl_bypass_resource_ids to az cosmosdb update/create
-Adds update serverversion
-Adds params backup_retention and backup_inteval to az cosmosdb update/craete
**Testing Guide**

- az cosmosdb create --name accountName --resource-group rgName --network-acl-bypass AzureServices --network-acl-bypass-resource-ids subscriptions/subId/resourceGroups/rgName/providers/Microsoft.Synapse/workspaces/wsName
- az cosmosdb update --name accountName --resource-group rgName --network-acl-bypass AzureServices --network-acl-bypass-resource-ids subscriptions/subId/resourceGroups/rgName/providers/Microsoft.Synapse/workspaces/wsName
- az cosmosdb update --name <account_name>--resource-group <resource_group> --server-version 4.0
-az cosmosdb create --name accountName --resource-group rgName --backup-interval 240 --backup-retention 8
-az cosmosdb update--name accountName --resource-group rgName --backup-interval 240 --backup-retention 8

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CosmosDB

**Comment by kristynhamasaki** [comment]
Hi @yonzhan @Juliehzl This change is needed for Ignite, could we change the label to S183? Thank you!

File: src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py [hunk 1]
```diff
@@ -75,6 +75,8 @@ def load_arguments(self, _):
             c.argument('disable_key_based_metadata_write_access', arg_type=get_three_state_flag(), help="Disable write operations on metadata resources (databases, containers, throughput) via account keys")
             c.argument('enable_public_network', options_list=['--enable-public-network', '-e'], arg_type=get_three_state_flag(), help="Enable or disable public network access to server.")
             c.argument('enable_analytical_storage', arg_type=get_three_state_flag(), help="Flag to enable log storage on the account.", is_preview=True)
+            c.argument('network_acl_bypass', options_list=['--network-acl-bypass'], help="Flag to enable or disable Network Acl Bypass.")
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py:78]: which value would be allowed to enable or disable acl bypass? If they are enum values, could you use `arg_type=get_enum_type(xxx)` here to show customer allowed values? If it is boolean, consider using `arg_type=get_three_state_flag()`.

File: src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py [hunk 2]
```diff
@@ -75,6 +75,8 @@ def load_arguments(self, _):
...
+            c.argument('network_acl_bypass_resource_ids', nargs='+', options_list=['--network-acl-bypass-resource-ids'], help="List of Resource Ids to allow Network Acl Bypass.")
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py:79]: could we add a shorter option for the parameter in your options list? e.g. `-i`.

Comment (kristynhamasaki) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py]: Done, thanks!

File: src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py (hunk 1 above)
Comment (kristynhamasaki) [reply src/azure-cli/azure/cli/command_modules/cosmosdb/_params.py]: It is an enum, I added get_enum_type(NetworkAclBypass)

**Comment by kristynhamasaki** [comment]
Hi @Juliehzl We have combined this change with 2 other CosmosDB changes. Please review

**Review by Juliehzl — APPROVED** [review]

**Comment by Juliehzl** [comment]
Hi @kristynhamasaki , could you fix CI failures?

**Comment by kristynhamasaki** [comment]
Hi @Juliehzl I reran the one test that hadn't been updated. Please queue the pipeline again. Thank you!

**Review by msyyc — APPROVED** [review]

**Pull request merged**